### ISSUE TITLE
reply_code and reply_text gets default values on disconnection

### ISSUE THREAD

**Issue by simmel** [pika/pika#382]
Using pika (0.9.13) with tornado (3.1) on Python 2.6.7 when I get disconnected using the official [Tornado example](https://pika.readthedocs.org/en/0.9.13/examples/tornado_consumer.html) I don't get the correct `reply_code` and `reply_text` but the default ones.

```
$ ./asynchronous_consumer_example.py
INFO       2013-07-25 12:36:07,976 __main__                       connect                              49  : Connecting to amqps://guest:guest@localhost:5671/%2F
INFO       2013-07-25 12:36:07,978 pika.adapters.base_connection  _create_and_connect_to_socket        164 : Connecting to 127.0.0.1:5671 with SSL
INFO       2013-07-25 12:36:08,108 __main__                       on_connection_open                   93  : Connection opened
INFO       2013-07-25 12:36:08,108 __main__                       add_on_connection_close_callback     64  : Adding connection close callback
INFO       2013-07-25 12:36:08,108 __main__                       open_channel                         310 : Creating a new channel
INFO       2013-07-25 12:36:08,112 __main__                       on_channel_open                      146 : Channel opened
INFO       2013-07-25 12:36:08,112 __main__                       add_on_channel_close_callback        118 : Adding channel close callback
INFO       2013-07-25 12:36:08,112 __main__                       setup_exchange                       159 : Declaring exchange message
INFO       2013-07-25 12:36:08,115 __main__                       on_exchange_declareok                171 : Exchange declared
INFO       2013-07-25 12:36:08,115 __main__                       setup_queue                          182 : Declaring queue text
INFO       2013-07-25 12:36:08,117 __main__                       on_queue_declareok                   196 : Binding message to text with example.text
INFO       2013-07-25 12:36:08,120 __main__                       on_bindok                            293 : Queue bound
INFO       2013-07-25 12:36:08,121 __main__                       start_consuming                      280 : Issuing consumer related RPC commands
INFO       2013-07-25 12:36:08,121 __main__                       add_on_cancel_callback               206 : Adding consumer cancellation callback
WARNING    2013-07-25 12:36:11,209 pika.adapters.base_connection  _check_state_on_disconnect           145 : Socket closed when connection was open
WARNING    2013-07-25 12:36:11,209 pika.connection                _on_disconnect                       1284: Disconnected from RabbitMQ at localhost:5671 (0): Not specified
WARNING    2013-07-25 12:36:11,210 __main__                       on_connection_closed                 82  : Connection closed, reopening in 5 seconds: (0) Not specified
INFO       2013-07-25 12:36:16,215 __main__                       connect                              49  : Connecting to amqps://guest:guest@localhost:5671/%2F
INFO       2013-07-25 12:36:16,217 pika.adapters.base_connection  _create_and_connect_to_socket        164 : Connecting to 127.0.0.1:5671 with SSL
INFO       2013-07-25 12:36:16,316 __main__                       on_connection_open                   93  : Connection opened
[...]
```

Before this method call https://github.com/pika/pika/blob/0.9.13/pika/adapters/base_connection.py#L124 it's correct and after it's wrong. I'm having trouble debugging the `_init_connection_state` method though.

**Comment by vitaly-krugl** [comment]
This is most likely fixed by PR #537

**Comment by simmel** [comment]
Sweet @vitaly-krugl. I can't test it I'm afraid, but I'll take your word for it.
